### PR TITLE
Wiki Inline-Commenting #1438

### PR DESCRIPTION
**Pull request by aspriya**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Picking up from where https://github.com/publiclab/plots2/pull/1571 left off.

### CODE REVIEW

**Comment by jywarren** [comment]
What's the error here? Can you copy it in? @ananyo2012 I wonder if danger
is responding to test failures but not errors? Or if an error happens early
enough maybe danger doesn't even run? If the latter is the case there's not
much we can do, but good to know.

**Comment by jywarren** [comment]
Ooh, cool - but see the warning about new schema file!

**Comment by jywarren** [comment]
I wanted to note that I think /all/ rich wiki updating is a bit slow. We'll look into this, but I believe we should:

* monitor unstable for performance (@icarito)
* add a "spinner" icon as with the Post button on `/post` (this also happens in regular comments, i think) so people know it's doing **something**
* fix the schema file as advised by Dangerbot above

then let's test one more time -- i'd like to work a bit more on how the comments are displayed too; perhaps we can show `3 comments` by the comment icons, instead of the full display shown in your screenshot:

image

Then that full display could be toggled to show.

File: app/assets/javascripts/buildSectionCommentForm.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+function buildSectionCommentForm(uniqueId, title, node_id, subsection_string) {
+  var formHtml = "<form style='display:none;' id='inline-comment-"+uniqueId+"' data-remote='true' class='well' action='/comment/create_inline_comment/"+node_id+"'>";
+  formHtml += "<h4 style='margin-top:0; padding-top:0;'>"+title+"</h4>";
```
Comment (jywarren) [inline app/assets/javascripts/buildSectionCommentForm.js]: Here I think you can do:

```js
var html = "<div>\
<p>Stuff</p>\
</div>";
```

File: app/assets/javascripts/showInlineComment.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  var iComment = "<div id='inline-comment-"+cid+"' style='padding-top: 10px; padding-left: 10px;'>";
+  iComment += "<div class='inline' style='vertical-align: top;'>";
+  if(photo_file_name){
+    iComment += "<img class='img-circle' src='"+author_photo_path+"' />";
+  }else{
+    iComment += "<div class='img-circle'></div>";
+  }
+  iComment += "</div>";
+
+  iComment += "<div class='inline' id='comment-content'>";
```
Comment (jywarren) [inline app/assets/javascripts/showInlineComment.js]: same here?

File: app/assets/javascripts/wikis.js [hunk 3]
```diff
@@ -15,7 +14,8 @@ function setupWiki(node_id, title, raw, logged_in) {
       preProcessor: preProcessMarkdown,
       postProcessor: postProcessContent,
       extraButtons: {
-        "fa-question": questionForm
+        "fa-question": questionForm,
+        "fa-comment": setupCommentFunction
```
Comment (jywarren) [inline app/assets/javascripts/wikis.js]: Can we make this:

```ruby
<% if params[:inlineCommenting] %>"fa-comment": setupCommentFunction<% end %>
```
? That way we can test this more on production without committing to it just yet. And we can publish the general Rich Wikis inline features without also having to publish inline commenting which may take a little longer?

File: db/schema.rb.example [hunk 4]
```diff
@@ -49,6 +49,7 @@ ActiveRecord::Schema.define(:version => 20170824172336) do
     t.string  "mail",      :limit => 64
     t.string  "homepage"
     t.integer "aid",                             :default => 0,  :null => false
+    t.string  "reference"
```
Comment (jywarren) [inline db/schema.rb.example]: Here I believe you have to update the timestamp at the top of the file to match the latest migration version -- that may be what's stopping you!

**Comment by icarito** [comment]
A note that not the whole site seems slow on unstable.publiclab.org - I
see no reason for slowness in the logs.

I will continue to investigate.

File: db/schema.rb.example (hunk 4 above)
Comment (aspriya) [reply db/schema.rb.example]: Oh thanx. Corrected this

File: app/assets/javascripts/buildSectionCommentForm.js (hunk 1 above)
Comment (jywarren) [reply app/assets/javascripts/buildSectionCommentForm.js]: Actually it looks like that syntax only works in node.js or maybe ES6; ignore!

File: app/assets/javascripts/wikis.js (hunk 3 above)
Comment (jywarren) [reply app/assets/javascripts/wikis.js]: Hi, @aspriya - i think if you can add this flag we can merge and see how this is working. Thanks!

**Comment by jywarren** [comment]
I made the "hidden" flag in javascript so this will only work when this is at the end of the URL: `?inlineComments=true`

Now all that's left is:

* [ ] add a "spinner" icon as with the Post button on /post (this also happens in regular comments, i think) so people know it's doing something

Very close!

**Comment by jywarren** [comment]
Hi, @Aspriya - what do you think about these changes? Hope you're well!

**Comment by ryzokuken** [comment]
@jywarren @aspriya hasn't replied to this thread for a very long time. Is this mergeable, or should we close this for now?

**Comment by jywarren** [comment]
do you want to download the code and try it out? I worried there were
things we overlooked in this and it needed a bit more work.

**Comment by ryzokuken** [comment]
@jywarren I could definitely give this a try!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**